ideascube/widgets.py

```diff
@@ -1,6 +1,56 @@
 from django.forms import widgets
 
 from ideascube.configuration import get_config
+from django.utils.html import mark_safe, escape
+from django.utils.encoding import force_text
+from django.forms.utils import flatatt
+
+
+_script_template = '''<script type="text/javascript">tinymce.init({{
+selector: "#{widget_id}",
+inline: true,
+menubar: {menubar},
+toolbar: {toolbar},
+plugins: "{plugins}",
+hidden_input: false
+}});</script>'''
+
+
+class RichTextEntry(widgets.Widget):
```

Comment:
We do not need to handle both. The `<input>` is just a hidden field filled by tinyMCE to lets HTML form and django properly send and handle the data in the post request.
The input area is replace by the div and the tinyMCE js modifying it.